server/migrations/helpers.js

```diff
@@ -1,50 +1,62 @@
 function calculatePreferences(user) {
-  const updatedPreferences = {
-    config: '',
-    sort: 0,
-    star: user?.preferences?.star || [],
-    complete: user?.preferences?.complete || [],
-  }
-  // Update complete
-  if (
-    !!user.preferences &&
-    user.preferences.complete &&
-    Object.keys(user.preferences.complete).length &&
-    !Array.isArray(user.preferences.complete)
-  ) {
-    updatedPreferences.complete = Object.keys(user.preferences.complete).map(
-      (siteName) => {
-        return {
-          site: siteName,
-          completedSubjects: user.preferences.complete[siteName],
-        }
+  const siteSubjects = []
+
+  if (!user.siteSubjects?.length) {
+    user.access.forEach((site, i) => {
+      const { preferences } = user
+      const siteSubject = {
+        site,
+        starredSubjects: preferences.star[site]?.length
+          ? preferences.star[site]
+          : [],
+        completedSubjects: preferences.complete[site]?.length
+          ? preferences.complete[site]
+          : [],
       }
-    )
+      siteSubjects.push(siteSubject)
+    })
   }
+  if (user.siteSubjects && user.siteSubjects.length) {
+    user.access.forEach((study, i) => {
+      const siteSubject = {
+        site: study,
+        starredSubjects: [],
+        completedSubjects: [],
+      }
+
+      if (user.siteSubjects.findIndex(({ site }) => study === site) > -1) {
```

Comment:
can we use `find` here instead of `findIndex`?

```suggestion
      if (user.siteSubjects.find(({ site }) => study === site)) {
```

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/bf0423c6c7f4d2b24d8c720cb39b4accdaccbbc7